### PR TITLE
Add yamlprocessor from PyPI

### PR DESCRIPTION
**Pull request by matthewrmshin**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by matthewrmshin** [comment]
I am willing to be listed in the maintainer section.

**Comment by KrissyRayD242** [comment]
I am willing to be listed in the maintainer section.

**Comment by matthewrmshin** [comment]
@conda-forge/help-python, ready for review!

File: recipes/yamlprocessor/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+  entry_points:
+    - yp-data = yamlprocessor.dataprocess:main
+    - yp-schema = yamlprocessor.schemaprocess:main
+
+requirements:
+  host:
+    - pip
+    - python >=3.7
```
Comment (BastianZim) [inline recipes/yamlprocessor/meta.yaml:23]:
```suggestion
    - python >=3.7
    - pytest-runner
```
Ref: https://github.com/JCSDA-internal/yamlprocessor/blob/1a323c0cad28bf47e2ed200e24ad5e9239a70805/setup.cfg#L40

File: recipes/yamlprocessor/LICENSE [hunk 2]
```diff
@@ -0,0 +1,29 @@
+BSD 3-Clause License
```
Comment (BastianZim) [inline recipes/yamlprocessor/LICENSE:1]: The license file is already included in the source distribution so you can remove it from here.

**Review by BastianZim — CHANGES_REQUESTED** [review]
Just a couple of small things, then this is ready to go 🚀

**Comment by matthewrmshin** [comment]
@BastianZim Thanks for reviewing. I have followed your recommendation in the latest change. 👍🏼

**Review by BastianZim — APPROVED** [review]

**Pull request merged**